main_logic/core.py

```
@@ -2352,81 +2355,98 @@ async def _process_stream_data_internal(self, message: dict):
             logger.error(f"💥 {error_message}")
             await self.send_status(error_message)
 
-    async def end_session(self, by_server=False):  # 与Core API断开连接
+    async def end_session(self, by_server=False, expected_session=None):  # 与Core API断开连接
         await self._init_renew_status()
 
         async with self.lock:
-            if not self.is_active:
+            if expected_session is not None and self.session is not expected_session:
+                logger.info("⏭️ end_session 跳过：session 已被新实例替换")
```

Ruff (0.15.4)

[warning] String contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF001)